patent.py

```diff
@@ -0,0 +1,40 @@
+import argparse
+import requests
+import json
+
+
+def patent(**kwargs):
+    patent_url = 'https://api.nasa.gov/patents/content'
+    get_params = {}
+    for key, value in kwargs.iteritems():
+        get_params[key] = value
+    get_response = requests.get(patent_url, params = kwargs)
+
+    if get_response.status_code == 429:
+        raise Exception('You have exceeded your rate limit')
+
+    body = get_response.json()
+
+    if 'error' in body:
+        raise Exception(body['error'])
+# Current default is to print out json, need to add ability for file output
```

Comment:
Remember, these functions will be consumed by other Python modules. We really aren't looking to interact with the user in any way here, but to return useful data that other Python code can consume. How about a dictionary?